packages/account/src/Components/unlink-account-modal/unlink-account-modal.tsx

```diff
@@ -1,9 +1,24 @@
 import React from 'react';
-import PropTypes from 'prop-types';
 import { Button, Modal, Text, Icon } from '@deriv/components';
 import { localize, Localize } from '@deriv/translations';
 
-const UnlinkAccountModal = ({ onClose, is_open, identifier_title, onClickSendEmail }) => {
+type TUnlinkAccountModalProps = {
+    onClose: () => void;
+    is_open: boolean;
+    identifier_title: string;
+    onClickSendEmail: () => void;
+};
+
+/**
+ * Modal displayed when user clicks on the 'Change email' button in the account settings page.
+ * @name UnlinkAccountModal
+ * @param {Function} onClose - function to close the modal
+ * @param {boolean} is_open - state to toggle the modal
+ * @param {string} identifier_title - title of the identifier (e.g. Google, Facebook)
+ * @param {Function} onClickSendEmail - function to send email to user
+ * @returns {React.ReactNode} - returns jsx component
```

Comment:
In my opinion, this kind of documentation especially for components is a bit too much, It will get outdated very quickly over time. I would suggest instead using TSDoc in the types.

```ts
type TProps = {
    /** Callback to be called once the dialog has been closed. */
    onClose: () => void;
    ...
};
```

Reply:
I preferred documenting props because the type may not be in the same file. When dev reds the component code they can see the prop description  as well

Reply:
@likhith-deriv Prop types of a component are isolated to the component itself and ideally, we won't be using them anywhere else. I've seen in the CFD package we have a huge prop type file for all the components and then export everything from there, We shouldn't be doing that 😬